### PR TITLE
Timeseries chart for R(t)

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Creates a new chart for a new metric we are developing, R(t) (rate of spread over time). As this is a statistical model we also show the confidence intervals. Features in common with the `CurveChart` (tooltip, axis styles) were factored out to the design system for reuse. 

Screen Shot 2020-04-23 at 3 21 08 PM

(There are some slight differences from the mocks due to some design issues that are still being worked out but they don't have any major implications for this PR, so no reason to hold it up IMO.)

I also included a container component that does some data transformation on some fake data, which I was using for development. I don't think the transformation logic is likely to change much once it's wired up, but if we think it's a bad idea to commit the dummy data to `master` I can take that out. (I'm of two minds; I think it makes this PR more comprehensible but if we were to somehow inadvertently release a UI to prod with dummy data that would obviously be bad! We have checked in dummy data before but now that we are getting into more of a release cadence maybe that's a practice we want to abandon.)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Part of #201 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally **(Verified no regressions in existing chart from refactoring)**

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/rt-timeseries/RtTimeseriesContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const convertToDate = isoParse;
+
+function dateExists(value: { date: Date | null }): value is Record {
+  // date formatting function may return null with invalid input
+  // so we do need a type guard to filter the results
+  return value.date != null;
+}
+
+const RtTimeseriesContainer: React.FC = () => {
+  // TODO: this is dummy data
```
Comment (jessex) [inline src/rt-timeseries/RtTimeseriesContainer.tsx:15]: Thanks for the thoughtful consideration of this in the PR description. I am fine placing this into a `__tests__` directory as a place where we could manually run the test, but I am becoming less comfortable with checking this kind of thing into master now that, as you said, release cadence is increasing, and user counts are climbing.

**Comment by macfarlandian** [comment]
@jessex I resurrected the TestPage locally in order to develop this but I don't know whether we want to check in that change either

**Comment by jessex** [comment]
Would rather not check that into master. However, I think it may be okay to have a process whereby for large, new changes that don't have a dedicated surface yet, we check in some brutalist test page for the sake of review, with LOUD, VISIBLE TODOs to not merge until it is removed. But I haven't thought this through very much. I don't think you should block this particular PR on it, in any event.

File: src/rt-timeseries/RtTimeseriesContainer.tsx (hunk 1 above)
Comment (macfarlandian) [reply src/rt-timeseries/RtTimeseriesContainer.tsx:15]: what if I just commented it out for now, with a louder TODO? putting random stuff in a `__tests__` directory will start causing problems with Jest once we get that up and running

Comment (jessex) [reply src/rt-timeseries/RtTimeseriesContainer.tsx:15]: Sure seems reasonable for now.

**Pull request merged**